### DISCUSSION TITLE
[IntelliSense] not working in turborepo package

### DISCUSSION THREAD

**Discussion by JasonAtallah** [tailwindlabs/tailwindcss#11347] [Ideas]
Intellisense in vscode doesn't work in turborepo packages.

# Steps to reproduce
1. Generate a new turborepo monorepo - `npx create-turbo@latest`.
2. Install tailwind, postcss and autoprefixer (either to the pre-generated "web" application or to the root).

# Expected result
3. Intellisense works in the "web" application and the "ui" package that is shared across apps.

# Actual result
3. Intellisense only works in the web app and not the ui package.

--

The details state, "In order for the extension to activate you must have tailwindcss installed and a Tailwind config file named tailwind.config.js or tailwind.config.cjs in your workspace."

I tried adding a tailwind config to the root and in the ui package and I still don't get intellisense in the ui package.

**Comment by VladymyrPylypchatin** [comment]
@JasonAtallah I had the same problem.

[Updating VS code settings as recommended in their official repo solved it.](https://github.com/tailwindlabs/tailwindcss-intellisense#extension-commands)

**Reply by datduyng** [reply]
The link is out of date. Does anyone have a more up to date link?

**Reply by robinvriens** [reply]
this worked, thanks!

**Reply by cchin25** [reply]
> @JasonAtallah I had the same problem.
> 
> [Updating VS code settings as recommended in their official repo solved it.](https://github.com/tailwindlabs/tailwindcss-intellisense#extension-commands)

For future reference,
```
    "tailwindCSS.experimental.configFile": {
        "packages/my-package/tailwind.config.ts": "packages/my-package/**"
    },
```
    
   Don't forget you may need to reload your VSCode window.

**Reply by stubinubin** [reply]
Anyone using v4 and only uses globals.css point to that file instead.
```
"tailwindCSS.experimental.configFile": {
        "packages/my-package/src/globals.css": "packages/my-package/**", # OR globally
        "packages/my-package/src/globals.css": "**"
    },

```

**Comment by a-eid** [comment]
was anybody able to get this to work ?

**Reply by stubinubin** [reply]
> Anyone using v4 and only uses globals.css point to that file instead.
> 
> ```
> "tailwindCSS.experimental.configFile": {
>         "packages/my-package/src/globals.css": "packages/my-package/**", # OR globally
>         "packages/my-package/src/globals.css": "**"
>     },
> ```

I use the global option, where my ui package is /packages/ui/src/globals.css

Finally, ensure you add the '@source' for where your `apps` are in globals.css.
```
@import 'tailwindcss';
@source "../../../apps/**/*.{ts,tsx}"; /* apps directory where you might have a web based ui. */
@source "../**/*.{ts,tsx}"; /* local ui package, contains common components. */
@import "tw-animate-css"
```

**Comment by feliperugai** [comment]
I have the same issue, but I got no intelissense at any apps/packages.

EDIT: Was on Pre-Release version, after downgrading to v.0.10.5 got it working.

**Comment by AdrianPaulV** [comment]
Same issue. I'm in a Turborepo monorepo. Intellisense works in apps/web and apps/web-pro but not in packages/ui. I've been trying to fix this for two days now. Going insane.

**Reply by programming-with-ia** [reply]
any solution?

**Reply by stubinubin** [reply]
> > Anyone using v4 and only uses globals.css point to that file instead.
> > ```
> > "tailwindCSS.experimental.configFile": {
> >         "packages/my-package/src/globals.css": "packages/my-package/**", # OR globally
> >         "packages/my-package/src/globals.css": "**"
> >     },
> > ```
> 
> I use the global option, where my ui package is /packages/ui/src/globals.css
> 
> Finally, ensure you add the '@source' for where your `apps` are in globals.css.
> 
> ```
> @import 'tailwindcss';
> @source "../../../apps/**/*.{ts,tsx}"; /* apps directory where you might have a web based ui. */
> @source "../**/*.{ts,tsx}"; /* local ui package, contains common components. */
> @import "tw-animate-css"
> ```

**Comment by rozsazoltan** [comment]
Reference:
* [VSCode Tailwind CSS Intellisense autocompletion and suggestions not working after installing Tailwind v4](https://stackoverflow.com/a/79433860/15167500)
* [Tailwind CSS Intellisense autocomplete does not work in React+Vite project](https://stackoverflow.com/a/79581167/15167500)
* https://stackoverflow.com/a/79551543/15167500
* https://discord.com/channels/486935104384532500/1436255793623339110/1436298809306255442
